intern/cycles/kernel/kernel_emission.h

```diff
@@ -87,15 +87,14 @@ ccl_device_noinline_cpu float3 direct_emissive_eval(KernelGlobals *kg,
       eval = shader_emissive_eval(emission_sd);
     }
   }
-
   eval *= ls->eval_fac;
 
   if (ls->lamp != LAMP_NONE) {
     const ccl_global KernelLight *klight = &kernel_tex_fetch(__lights, ls->lamp);
     eval *= make_float3(klight->strength[0], klight->strength[1], klight->strength[2]);
   }
 
-  return eval;
+  return linear_to_wavelength_intensities(eval, state->wavelengths);
```

Comment:
What has had to happen to convert cycles is to convert the RGB values (here represented by the floats in `klight->strength`) passed into various BSDFs to the intensities of the wavelengths being sampled (in `state->wavelengths`). This can be thought of a two step process.

Step 1. The BSDF (here evaluating direct emission) is given an RGB value to multiply the intensity by. We convert this to a spectrum with the sRGB to spectrum curves you might have seen on the devtalk thread.
Step 2. We sample this spectrum at the 3 wavelengths in the current ray.

These two steps are collapsed together into `linear_to_wavelength_intensities` for performance reasons, and because I don't know how to make a function and store that in a variable to be called later, and even if I did, it seems like a huge performance overhead.